src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ImportMetadataFunctionTests.cs

```diff
@@ -156,6 +157,150 @@ public async Task DataSetVersionMeta_CorrectLocations(ProcessorTestData testData
+        public async Task DataSetVersionMeta_CorrectLocationOptions_WithMappings(
+            ProcessorTestData testData,
+            int metaInsertBatchSize)
+        {
+            var (sourceDataSetVersion, _) = await CreateDataSet(Stage.PreviousStage());
+
+            var (targetDataSetVersion, instanceId) = await CreateDataSetVersionAndImport(
+                dataSetId: sourceDataSetVersion.DataSet.Id,
+                importStage: DataSetVersionImportStage.ManualMapping,
+                versionMinor: 1);
+
+            // In this test, we will create mappings for all the original location options.
+            // 2 of these mappings will have candidates, and the rest will have no candidates
+            // mapped.
+
+            // Amend a couple of arbitrary mappings to identify some candidates.
+            var mappedOption1 = testData.ExpectedLocations.First().Options.First();
+            var mappedOption2 = testData.ExpectedLocations.Last().Options.Last();
+
+            var option1Mapping = new LocationOptionMapping
+            {
+                PublicId = "option-1-public-id",
+                Type = MappingType.AutoMapped,
+                CandidateKey = MappingKeyFunctions.LocationOptionMetaRowKeyGenerator(mappedOption1.ToRow())
+            };
+
+            var option2Mapping = new LocationOptionMapping
+            {
+                PublicId = "option-2-public-id",
+                Type = MappingType.ManualMapped,
+                CandidateKey = MappingKeyFunctions.LocationOptionMetaRowKeyGenerator(mappedOption2.ToRow())
+            };
+
+            var random = new Random();
+
+            var mappings = new DataSetVersionMapping
+            {
+                SourceDataSetVersionId = sourceDataSetVersion.Id,
+                TargetDataSetVersionId = targetDataSetVersion.Id,
+                FilterMappingPlan = new FilterMappingPlan(),
+                LocationMappingPlan = new LocationMappingPlan
+                {
+                    Levels = testData
+                        .ExpectedLocations
+                        .ToDictionary(
+                            keySelector: level => level.Level,
+                            elementSelector: level => new LocationLevelMappings
+                            {
+                                Mappings = level
+                                    .Options
+                                    .ToDictionary(
+                                        keySelector: MappingKeyFunctions.LocationOptionMetaKeyGenerator,
+                                        elementSelector: option =>
+                                            option == mappedOption1 ? option1Mapping
+                                            : option == mappedOption2 ? option2Mapping
+                                            : new LocationOptionMapping
+                                            {
+                                                Type = random.Next(1) == 0
```

Comment:
I think `random.Next(1)` will always produce `0`

Reply:
wonder if it would be better to make this deterministic, and not use `random`. Declare our own integer, and take the modulus so that it switches between `AutoNone` and `ManualNone` for consecutive options

Reply:
Good idea - implemented this thanks